pupyl/cli.py

```diff
@@ -0,0 +1,13 @@
+""" Manage cli arguments"""
+
+import argparse
+
+
+class PupylCommandLineInterface:
+    def __init__(self):
+        self.parser = argparse.ArgumentParser(
+            description='Manage pupyl cli arguments.'
+        )
+        self.parser.add_argument(
+            '--data_dir', help='data directory for image files'
+        )
```

Comment:
@92amartins, the argument `data_dir` should be positional, not optional. Maybe it's better:

```suggestion
        self.parser.add_argument(
            'data_dir', help='Data directory for image files.'
        )
```

There's an idea to let `data_dir` as an optional argument, but we should clearly advise (as in [README.md](https://github.com/policratus/pupyl#-usage)) the user that in this case (of not filling this argument) all assets will be created on their temporary directory (and probably will be lost on next reboot).

So, for instance:

```shell
# Creates all assets on the system temporary directory
pupyl --index /path/to/images
```
or
```shell
# Creates all asset on /path/to/assets
pupyl --index /path/to/images --data_dir /path/to/assets
```

Reply:
For the sake of simplicity, I think we should go for mandatory, but support `data_dir` as a optional argument in the future.

Reply:
Regarding the suggestion on the help text, I just followed the `-h` defaults (no uppercase, no end stop).
I think we should stick to it.

positional arguments:
  data_dir    data directory for image files

optional arguments:
  -h, --help  show this help message and exit

Reply:
Lastly, could you clarify on the difference between `/path/to/images` and `/path/to/assets`? My understanding is that the first refers to _input images_ and the second is some kind of intermediate files for pupyl's internal representation.

Reply:
> For the sake of simplicity, I think we should go for mandatory, but support `data_dir` as a optional argument in the future.

Totally agree! 🚀

Reply:
> Regarding the suggestion on the help text, I just followed the `-h` defaults (no uppercase, no end stop).
> I think we should stick to it.
> 
> ```
> positional arguments:
>   data_dir    data directory for image files
> 
> optional arguments:
>   -h, --help  show this help message and exit
> ```

You're right, my bad!

Reply:
> Lastly, could you clarify on the difference between `/path/to/images` and `/path/to/assets`? My understanding is that the first refers to _input images_ and the second is some kind of intermediate files for pupyl's internal representation.

Yep, kinda.

**TL;DR**
------------
**`/path/to/images` is like a directory with CSVs to use as input data for a database; `/path/to/assets` is where the database internal files are stored.**

**TS:WM**
------------
The path reference used on [pupyl.search.PupylImageSearch.index()](https://github.com/policratus/pupyl/blob/c79f68d2d1ab0651e0a0316da963da8687315cc2/pupyl/search.py#L105) (and other subsequent methods) describes a _source_, or in other words the _input data_ or the images that must be inserted and indexed. Hence, references to the example path `/path/to/images` are exactly this, image that will be read.

References to `data_dir` (as in [pupyl.search.PupylImageSearch.__init__(self, data_dir=None, **kwargs)](https://github.com/policratus/pupyl/blob/c79f68d2d1ab0651e0a0316da963da8687315cc2/pupyl/search.py#L29l)) are related to internal representations (and files) of a created database, which contains copies of `/path/to/images` images, index files, metadata, etc.

Reply:
Nice. Thanks for the clarifications o/